commons/src/main/java/org/eclipse/kapua/commons/util/xml/JAXBContextProviderImpl.java

```diff
@@ -12,45 +12,48 @@
  *******************************************************************************/
 package org.eclipse.kapua.commons.util.xml;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-import javax.xml.bind.JAXBContext;
-import javax.xml.bind.JAXBException;
-
 import org.eclipse.kapua.KapuaException;
 import org.eclipse.kapua.commons.core.ClassProvider;
 import org.eclipse.persistence.jaxb.JAXBContextFactory;
 import org.eclipse.persistence.jaxb.MarshallerProperties;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class JAXBContextProviderBase implements JAXBContextProvider {
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Singleton
```

Comment:
Normally you want to have only of these around, to perform the costly class path scan only once.